### PR TITLE
ArrayFire ebuild

### PR DESCRIPTION
**Pull request by marbre**
Add ebuild for arrayfire to the overlay.

OpenCL is not yet supported and therefore deactivated.

### CODE REVIEW

File: licenses/ArrayFire [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2014, ArrayFire
+All rights reserved.
```
Comment (jlec) [inline licenses/ArrayFire]: this is one of the BSD* licenses. Please check for the correct one.

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+DEPEND="${RDEPEND}
+	test? ( dev-vcs/subversion )"
+
+S="${WORKDIR}/${P}"
+BUILD_DIR="${S}/build"
+CMAKE_BUILD_TYPE=Release
+
+# We need write acccess /dev/nvidiactl, /dev/nvidia0 and /dev/nvidia-uvm and the portage
+# user is (usually) not in the video group
+if use cuda; then
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Drop the if then clause and add the RESTRICT USE independent.

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	virtual/blas
+	virtual/cblas
+	cuda? ( >=dev-util/nvidia-cuda-toolkit-6.0 )
+	sci-libs/fftw:3.0"
+DEPEND="${RDEPEND}
+	test? ( dev-vcs/subversion )"
+
+S="${WORKDIR}/${P}"
+BUILD_DIR="${S}/build"
+CMAKE_BUILD_TYPE=Release
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Leave these two alone. They are handled by the cmake-utils.eclass

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+RDEPEND="
+	>=sys-devel/gcc-4.7.3-r1
+	virtual/blas
+	virtual/cblas
+	cuda? ( >=dev-util/nvidia-cuda-toolkit-6.0 )
+	sci-libs/fftw:3.0"
+DEPEND="${RDEPEND}
+	test? ( dev-vcs/subversion )"
+
+S="${WORKDIR}/${P}"
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: This is the default, so you can drop it.

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+IUSE="+examples +cpu cuda test"
+
+RDEPEND="
+	>=sys-devel/gcc-4.7.3-r1
+	virtual/blas
+	virtual/cblas
+	cuda? ( >=dev-util/nvidia-cuda-toolkit-6.0 )
+	sci-libs/fftw:3.0"
+DEPEND="${RDEPEND}
+	test? ( dev-vcs/subversion )"
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Why is this needed? Smells like fetching from the internet during build. This is not allowed.

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+HOMEPAGE="http://www.arrayfire.com/"
+EGIT_REPO_URI="https://github.com/arrayfire/arrayfire.git"
+KEYWORDS="~amd64"
+
+LICENSE="ArrayFire"
+SLOT="0"
+IUSE="+examples +cpu cuda test"
+
+RDEPEND="
+	>=sys-devel/gcc-4.7.3-r1
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: If you add this dependency you need to test for the compiler during pkg_pretend()

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+inherit cmake-utils eutils git-r3
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: no need to inherit eutils

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+fi
+
+src_prepare() {
+	if use cpu; then
+		epatch "${FILESDIR}/FindCBLAS.patch"
+	fi
+	if use examples; then
+		epatch "${FILESDIR}/CMakeLists_examples.patch"
+	fi
+
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Patching should be USE independent. You can use the PATCHES array here.

File: sci-libs/arrayfire/arrayfire-9999.ebuild [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	   $(cmake-utils_use_build test TEST)
+	)
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	exeinto /usr/bin
+	doexe "build/bin2cpp"
```
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: simply use dobin here.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 3 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: CMAKE_BUILD_TYPE=Release is necessary. Otherwise it is set to gentoo and therefore the -DBUILD_TEST=on fails. It took hours to find this out ;)
I'll check for the BUILD_DIR setting.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 5 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: arrayfire fetches a svn repo for to googletests target. How should we handle this, if fetching during the build is not allowed?

Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: do a snapshot and fetch it with test? ( .. ). If you need to have it mirrored, I can place it on my devspace.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 3 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: In general it is possible to let gentoo handle the BUILD_DIR. Unfortunately I get in trouble with the new src_unpack phase. Therefore I think it is okay to leave it as is.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 4 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Done.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 5 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Download gtest via SRC_URI (if test is enable) and unpack it in src_unpack.

File: licenses/ArrayFire (hunk 1 above)
Comment (marbre) [inline licenses/ArrayFire]: Done.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 6 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Done.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 7 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Dropped it.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 8 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Done.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 9 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Done.

File: licenses/ArrayFire (hunk 1 above)
Comment (jlec) [inline licenses/ArrayFire]: Drop this file from license dir.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 5 above)
Comment (jlec) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: so drop svn here.

File: sci-libs/arrayfire/arrayfire-9999.ebuild (hunk 2 above)
Comment (marbre) [inline sci-libs/arrayfire/arrayfire-9999.ebuild]: Done.

**Comment by marbre** [comment]
The new commits should fix all the issues. Thank you very much for the advices.
